include/swift/AST/Requirement.h

```diff
@@ -43,8 +46,11 @@ enum class RequirementKind : unsigned {
 /// \brief A single requirement placed on the type parameters (or associated
 /// types thereof) of a
 class Requirement {
-  llvm::PointerIntPair<Type, 2, RequirementKind> FirstTypeAndKind;
-  Type SecondType;
+  llvm::PointerIntPair<Type, 3, RequirementKind> FirstTypeAndKind;
+  union {
```

Comment:
I don't have a strong opinion. We can do it now, but should we add more alternatives in the future and have e.g. 6 different kinds of a payload, we probably don't want to use a PointerUnion for that, or? Plus, each getter already asserts if the tag is wrong.